web/src/features/time/HistoricalTimeHeader.tsx

```diff
@@ -0,0 +1,102 @@
+import useGetState from 'api/getState';
+import Badge from 'components/Badge';
+import { Button } from 'components/Button';
+import { FormattedTime } from 'components/Time';
+import { useAtomValue } from 'jotai';
+import { ChevronLeft, ChevronRight } from 'lucide-react';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import { twMerge } from 'tailwind-merge';
+import { RouteParameters } from 'types';
+import { useNavigateWithParameters } from 'utils/helpers';
+import { endDatetimeAtom, startDatetimeAtom } from 'utils/state/atoms';
+
+function setToMidnightUTC(date: Date): Date {
+  const newDate = new Date(date);
+  newDate.setUTCHours(0, 0, 0, 0);
+  return newDate;
+}
+function formatDateToISO(date: Date): string {
+  return date.toISOString().slice(0, -5) + 'Z';
+}
+
+export default function HistoricalTimeHeader() {
+  const { i18n } = useTranslation();
+  const startDatetime = useAtomValue(startDatetimeAtom);
+  const endDatetime = useAtomValue(endDatetimeAtom);
+  const { isLoading } = useGetState();
+  const { urlDatetime } = useParams<RouteParameters>();
+  const navigate = useNavigateWithParameters();
+  function handleLeftClick() {
+    if (!urlDatetime && !endDatetime) {
+      return;
+    }
+
+    const targetDate = urlDatetime ? new Date(urlDatetime) : endDatetime;
+    if (!targetDate) {
+      return;
+    }
+
+    const midnightDate = setToMidnightUTC(targetDate);
+    const newDate = new Date(midnightDate);
+    urlDatetime
+      ? newDate.setUTCDate(midnightDate.getUTCDate() - 1)
+      : newDate.setUTCDate(midnightDate.getUTCDate());
+
+    navigate({ datetime: formatDateToISO(newDate) });
+  }
+
+  function handleRightClick() {
+    if (!endDatetime) {
+      return;
```

Comment:
nit: Do we want to display some sort of message to the user if clicking the historical direction buttons fail?

Reply:
I've improved the buttons so hopefully the isDisabled flag can communicate that there's no expected action